File: book-app/BookApp-Frontend/src/app/models/Book.ts
export class Book{

    private _bookName:string;
    private _authorName:string;
        private _description:string;
        private _imageUrl:string;
        private _price:string 

    constructor(
        bookName:string,
        authorName:string,
        description:string,
        imageUrl:string,
        price:string 
        ){

            this._bookName =bookName;
        this._authorName =authorName;
        this._description =description;
        this._imageUrl =imageUrl;
        this._price =price 

        }

//getters
get name() : string {
    return this._bookName;
}

public get authorName() : string {
    return this._authorName;
}

public get description() : string {
    return this._description;
}

public get imageUrl() : string {
    return this._imageUrl;
}

public get price() : string {
    return this._price;
}

//setters
public set name(name : string) {
    this._bookName = name;
}

public set authorName(authorName : string) {
    this._authorName = authorName;
}

public set description(description : string) {
    this._description = description;
}

public set imageUrl(imageUrl : string) {
    this._imageUrl = imageUrl;
}

public set price(price : string) {
    this._price = price;
}



}

File: book-app/BookApp-Backend/book-service/src/main/java/in/bookapp/bookservice/model/Book.java
package in.bookapp.bookservice.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.MongoId;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "books")
public class Book {

    @MongoId
    @JsonSerialize(using = ToStringSerializer.class)
    private ObjectId id;

    private String bookName;

    private String authorName;

    private String description;

    private String imageUrl;

    private String price;

    public Book(String bookName, String authorName, String description, String imageUrl, String price) {
        this.bookName = bookName;
        this.authorName = authorName;
        this.description = description;
        this.imageUrl = imageUrl;
        this.price = price;
    }

} 
